gatherers/manifest.js

```diff
@@ -54,34 +38,87 @@ class Manifest extends Gather {
     };
   }
 
+  /**
+   * @private
+   */
+  static _createDeferred() {
+    let resolve;
+    let reject;
+    const promise = new Promise((internalResolve, internalReject) => {
```

Comment:
FYI this is a super anti-pattern and allows for errors to happen without being caught, I would like to see us not using this pattern if at all possible.